scripts/scripts.js

```diff
@@ -714,7 +714,76 @@ async function loadLazy(doc) {
     setTimeout(() => {
       const element = hash ? main.querySelector(hash) : false;
       if (element) {
-        element.scrollIntoView();
+        // Handle lazy-loaded images that might cause content shift
+        const handleLazyLoadedImages = () => {
+          // Find all images above the target element that are lazy-loaded
+          const allElements = Array.from(main.querySelectorAll('*'));
+          const targetIndex = allElements.indexOf(element);
+
+          if (targetIndex === -1) {
+            element.scrollIntoView();
+            return;
+          }
+
+          // Get all lazy-loaded images before the target element
+          const precedingImages = allElements
+            .slice(0, targetIndex)
+            .filter((el) => {
+              if (el.tagName === 'IMG' && el.getAttribute('loading') === 'lazy') {
+                return true;
+              }
+              // Also check for images inside picture elements
+              if (el.tagName === 'PICTURE') {
+                const img = el.querySelector('img[loading="lazy"]');
+                return img !== null;
+              }
+              return false;
+            })
+            .map((el) => (el.tagName === 'PICTURE' ? el.querySelector('img') : el));
```

Comment:
I think all this could be simplified to:
```suggestion
          const allImages = Array.from(document.querySelectorAll('img'));
          const precedingImages = allImages
            .filter(img => img.compareDocumentPosition(element) & Node.DOCUMENT_POSITION_PRECEDING;
        });
```
Like this, there also shouldn't be a need for special handling of `img` elements inside `picture` elements.